### PR TITLE
refact(hostsensorutils): refactors the host sensor

### PR DESCRIPTION
**Pull request by fredbi**
## Overview
This PR factorizes the list of calls to the host-scanner API in a loop.

More godoc-friendly doc strings are added.

### CODE REVIEW

**Comment by fredbi** [comment]
@matthyx @dwertent  this PR takes back the refactoring ideas present in #1066, but is less ambitious and brings fewer changes.

The idea is to factorize the boilerplate calls to the host-scanner API, without any further attempt to improve the sensor client.

@matthyx  I got your message regarding the difficulty to review too large PRs. So this one I've kept concise.

**Review by dwertent — APPROVED** [review]
Nicely done :)

File: core/pkg/hostsensorutils/hostsensorgetfrompod.go [hunk 1]
```diff
@@ -117,40 +109,42 @@ func (hsh *HostSensorHandler) GetVersion() (string, error) {
 		resBytes, err := hsh.HTTPGetToPod(job.podName, job.path)
 		if err != nil {
 			return "", err
-		} else {
-			version := strings.ReplaceAll(string(resBytes), "\"", "")
-			version = strings.ReplaceAll(version, "\n", "")
-			return version, nil
 		}
+
```
[the remark this answers is not in the document]
Comment (fredbi) [reply core/pkg/hostsensorutils/hostsensorgetfrompod.go:117]: reverted to the original if-else construct.

**Review by matthyx — APPROVED** [review]
@dwertent I think we wanted to merge some PRs last week, this one included, WDYT?

**Review by matthyx — APPROVED** [review]

**Pull request merged**